### PR TITLE
Add pyshex

### PR DESCRIPTION
**Pull request by jfy133**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pyshex/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/PyShEx-{{ version }}.tar.gz
+  sha256: 3c5c4d45fe27faaadae803cb008c41acf8ee784da7868b04fd84967e75be70d0
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
```
Comment (jfy133) [inline recipes/pyshex/meta.yaml:16]:
```suggestion
  entry_points:
    - generate_parser = pyshex.shex_evaluator:evaluate_cli

```

File: recipes/pyshex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/PyShEx-{{ version }}.tar.gz
+  sha256: 3c5c4d45fe27faaadae803cb008c41acf8ee784da7868b04fd84967e75be70d0
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  entry_points:
+    - generate_parser = pyshex.shex_evaluator:evaluate_cli
```
Comment (ocefpaf) [inline recipes/pyshex/meta.yaml:17]: I don't see that `generate_parser`. This one is in the metadata though:

```suggestion
    - shexeval = pyshex.shex_evaluator:evaluate_cli
```

Comment (jfy133) [reply recipes/pyshex/meta.yaml]: Oops, copy paste error, sorry!

**Comment by jfy133** [comment]
@ocefpaf Thank you once again :pray:

**Pull request merged**